### PR TITLE
MAINT: Add api-token for CircleCI redirector action

### PR DESCRIPTION
**Pull request by melissawm**
## PR Summary

The CircleCI artifact redirector action (which creates the docs preview links in PRs) [is failing](https://github.com/matplotlib/matplotlib/actions/runs/4681038876). This is due to [an upstream bug](https://github.com/larsoner/circleci-artifacts-redirector-action/issues/40). Following the advice on that thread, this PR adds the `api-token` definition in the configuration file for the action.

## PR Checklist

**Documentation and Tests**
- [N/A] Has pytest style unit tests (and `pytest` passes)
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] New plotting related features are documented with examples.

**Release Notes**
- [N/A] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [N/A] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [N/A] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: .github/workflows/circleci.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ jobs:
         uses: larsoner/circleci-artifacts-redirector-action@master
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
+          api-token: ${{ secrets.CIRCLE_TOKEN }}
```
Comment (tacaswell) [inline .github/workflows/circleci.yml:16]:
```suggestion
          api-token: ${{ secrets.CIRCLECI_TOKEN }}
```

There was an inconsistency in the naming in the instructions, I copied the name from step 3 you copied the name from step 2!

Comment (tacaswell) [reply .github/workflows/circleci.yml:16]: image

Comment (tacaswell) [reply .github/workflows/circleci.yml:16]: I can not rename it without fully regenerating it 😞

**Comment by jklymak** [comment]
I don't know if that helps or not - we are failing way earlier when we try and install codecov?

**Comment by melissawm** [comment]
I believe that is a separate issue - see https://community.codecov.com/t/codecov-yanked-from-pypi-all-versions/4259 and https://matrix.to/#/!BXmyZMTnRjWJldDRLV:gitter.im/$ymSy4MqndKDoNUxv93H64cNINSloJBMNr0oWAZ5fbMU?via=gitter.im&via=matrix.org&via=aria-net.org

**Comment by QuLogic** [comment]
Can you drop codecov/coverage here? Re: https://matrix.to/#/!BXmyZMTnRjWJldDRLV:gitter.im/$_KoAHkkUoHKgNSvjo0ByDF0XZWwSXJaErOodoYzi8AQ?via=gitter.im&via=matrix.org&via=aria-net.org

**Comment by melissawm** [comment]
Ok - I think this is ready (I don't think the CI failures are related). 

This will only work once merged to main - you can see similar work done here: https://github.com/scipy/scipy/pull/18292

**Comment by rcomer** [comment]
@meeseeksdev backport to v3.7.1-doc

**Pull request merged**